### PR TITLE
fix(structured-list): implement keyboard navigation for new markup

### LINKED ISSUE #1426: Broken keyboard navigation for StructuredList with selection
## Detailed description

Full keyboard navigation is not available for the StructuredList with selection

> What did you expect to happen? What happened instead? What would you like to see changed?

While the arrow keys will change the focus when attempting to select different items, the actual selection is not performed when the space bar or Enter key is pressed. It also appears that the arrow keys are also attempting to make selections when pressed, but that is not functioning properly either

> What browser are you working in?

Firefox 65+, Chrome 70+

> What version of the Carbon Design System are you using?

9.58.3

## Steps to reproduce the issue

1. tab into a structured list
2. navigate the list with tab or arrow keys
3. press Enter or space bar to make new selection

## Additional information

Related https://github.com/IBM/carbon-components-react/issues/1555

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1426

This PR will fix the keyboard navigation behavior for structured lists, in accordance to http://w3c.github.io/aria-practices/#radiobutton

> * When a radio group receives focus:
>   * If a radio button is checked, focus is set on the checked button.
>   * If none of the radio buttons are checked, focus is set on the first radio button in the group.
> * `Space`: checks the focused radio button if it is not already checked.
> * `Right Arrow` and `Down Arrow`: move focus to the next radio button in the group, uncheck the previously focused button, and check the newly focused button. If focus is on the last button, focus moves to the first button.
> * `Left Arrow` and `Up Arrow`: move focus to the previous radio button in the group, uncheck the previously focused button, and check the newly focused button. If focus is on the first button, focus moves to the last button.

#### Changelog

**New**

- event handlers for left/right arrow keydown events

**Changed**

The previous `keydown` handlers were looking for `<label>` rows with a `for` attribute for a corresponding labelable element, but the `for` attribute is not required when using a `<label>` due to the implicit association by nesting the corresponding `input` in our templates. The new element selection query will:

- in the case of `Enter` or `Space` keypress: find the nested `input` in the selected row
- in the case of arrow key keypress: find the nested `input` in the row with the correct index that the user is navigating towards

- handle enter or space `keydown` events for selecting `<label>` rows
- handle up arrow and down arrow `keydown` events for selecting `label` rows

#### Testing / Reviewing

1. tab into a structured list
2. navigate the list with tab or arrow keys
3. press Enter or space bar to make new selection

related issues:

https://github.com/IBM/carbon-components-react/issues/1555

https://github.com/IBM/carbon-components-react/pull/1528

### CODE REVIEW

File: src/components/structured-list/structured-list.js [hunk 1]
```diff
@@ -47,13 +49,9 @@ class StructuredList extends mixin(createComponent, initComponentBySearch, handl
...
   }
 
-  _getInput(index) {
-    const rows = [...this.element.querySelectorAll(this.options.selectorRow)];
-    return this.element.ownerDocument.querySelector(this.options.selectorListInput(rows[index].getAttribute('for')));
-  }
-
   _handleInputChecked(index) {
-    const input = this._getInput(index);
+    const rows = [...this.element.querySelectorAll(this.options.selectorRow)];
```
Comment (asudoh) [inline src/components/structured-list/structured-list.js]: Spread operator here won't be needed as ondex reference of `NodeList` should work.

**Review by asudoh — APPROVED** [review]
LGTM 👍, just one thing - Thanks @emyarod!

File: src/components/structured-list/structured-list.js (hunk 1 above)
Comment (emyarod) [reply src/components/structured-list/structured-list.js]: thanks for pointing that out, I just moved it over from the `_getInput` method

**Comment by emyarod** [comment]
@asudoh just double checking: does the logic for changing the checked input look alright? do we need to explicitly set `input.checked` to `true` in `_handleClick`?

**Review by asudoh — APPROVED** [review]
LGTM :+1: - Thanks @emyarod for the update!

**Review by alisonjoseph — APPROVED** [review]
Thanks for updating this! LGTM

**Pull request merged**